File: sdk/runanywhere-kotlin/src/jvmTest/kotlin/com/runanywhere/sdk/perf/CancelParityTest.kt
/**
 * CancelParityTest.kt — JUnit runner for GAP 09 #7 (v3.1 Phase 5.1).
 */

package com.runanywhere.sdk.perf

import org.junit.Test
import org.junit.Assume
import java.io.File
import tests.streaming.cancel_parity.CancelParity
import kotlin.test.assertTrue
import kotlin.test.assertNotNull

class CancelParityTest {
    @Test
    fun `cancel parity records interrupt and writes trace`() {
        Assume.assumeTrue(
            "cancel_parity input missing at /tmp/cancel_input.bin",
            File(CancelParity.DEFAULT_INPUT_PATH).exists()
        )
        val result = CancelParity.run()
        assertTrue(result.total > 0, "expected >0 events")
        assertNotNull(result.interruptOrdinal, "expected interrupt marker")
    }
}

File: sdk/runanywhere-flutter/packages/runanywhere/test/cancel_parity_test.dart
// cancel_parity_test.dart — flutter_test runner for GAP 09 #7 (v3.1 Phase 5.1).

import 'dart:io';

import 'package:flutter_test/flutter_test.dart';

import '../../../../../tests/streaming/cancel_parity/cancel_parity.dart';

void main() {
  group('cancel_parity (dart)', () {
    setUpAll(() {
      if (!File(CancelParity.defaultInputPath).existsSync()) {
        fail(
          'cancel_parity input missing at ${CancelParity.defaultInputPath}',
        );
      }
    });

    test('records interrupt ordinal and writes trace', () async {
      final result = await CancelParity.run();
      expect(result.total, greaterThan(0));
      expect(result.interruptOrdinal, isNotNull);
    });
  });
}
